src/components/dialog/useEscapeKey.tsx

```diff
@@ -0,0 +1,19 @@
+import { useEffect } from "react";
+
+const useEscapeKey = ({ closeDialog }: { closeDialog: () => void }) => {
```

Comment:
Consider replacing this/some of this with the hooks provided by react-use.

We already have it in the project https://github.com/streamich/react-use?tab=readme-ov-file